ut/columns_ut.cpp

```diff
@@ -185,6 +185,17 @@ TEST(ColumnsCase, Date_UInt16_interface) {
     ASSERT_EQ(col1->RawAt(1), 1234u);
 }
 
+TEST(ColumnsCase, Date_UInt16_construct_from_rvalue_data) {
+    auto const expected = MakeNumbers<uint16_t>();
+
+    auto data = expected;
+    auto col1 = std::make_shared<ColumnDate>(std::move(data));
+
+    ASSERT_EQ(col1->Size(), expected.size());
+    for (size_t i = 0; i < expected.size(); ++i) {
+        ASSERT_EQ(col1->RawAt(i), expected[i]);
```

Comment:
`CompareRecursive` isn't going to work due to value transformation by `ColumnDate`, same for `DateTime32`. methods returning bogus `time_t` values is the hard legacy we need to get rid of, but I haven't decided how to make a smooth transition for the users...